Fix loading of I18n in plugins from Angular application; Reorganise frontend code

myabc opened the pull request:
:warning: **Please review and merge opf/openproject#2193 first.**

https://community.openproject.org/work_packages/5697
https://community.openproject.org/work_packages/14679

See also finnlabs/openproject-backlogs#114

myabc committed:
Add npm module skeleton; load plugin translations

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts commented on index.js:
```diff
@@ -0,0 +1,5 @@
+console.log('Costs loaded');
```
Please remove

ulferts commented:
We currently have some angular code in https://github.com/finnlabs/openproject-costs/tree/dev/app/assets/javascripts/angular. Just for me to understand: That would then have to be referenced here in order to initialise it? Why the change from the existing behaviour?

myabc committed:
Move Angular app plugin code under frontend/app

Analog to core commit:
https://github.com/opf/openproject/commit/6ec5ca58edc6d8dbf0efd14aab189dead0da6fa2

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
No longer load Angular code via asset pipeline

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@ulferts I went ahead and moved files in 414327a…

> We currently have some angular code in https://github.com/finnlabs/openproject-costs/tree/dev/app/assets/javascripts/angular. Just for me to understand: That would then have to be referenced here in order to initialise it? Why the change from the existing behaviour?

Moving this code wasn't strictly necessary _yet_. We could have continued to let Sprockets/ Rails asset pipeline take care of the bundling of this code.

However, if we want to achieve the goal of making front-end testable without Rails (including getting Protractor to test plugin front-end code), then it's worth bringing forward these changes.
